asllib/ASLTypingReference.mld

```diff
@@ -1073,49 +1172,136 @@ Typing of Blocks
 }
 
   {3 TypingRule.Block}
-
+    
     {4 Rule}
-    [annotate_block env stm] is <description>.
+    [annotate_block env return_type s] is the result of annotating the
+    statement [s] in [env].
+
+    {4 Info}
+    A local identifier declared with var, let or constant is in scope from the
+    point immediately after its declaration until the end of the immediately
+    enclosing block.
 
+    From that follows that we can discard the environment at the end of an
+    enclosing block.
+    
     {4 Example}
-    <minimal example>
 
 {2
 Typing of Catchers
 }
-[annotate_catchers env catchers otherwise_opt s_m]
+  
+  {3 TypingRule.CatcherNone}
+  [annotate_catchers env return_type (name_opt, ty, stmt)] is [(name_opt, ty,
+  new_stmt)] and one of the following applies:
+ 
+    {4 Rule}
+    All of the following applies:
+    - [ty] has the structure of an exception type;
+    - [name_opt] gives no name;
+    - [env'] is [env];
+    - [new_stmt] is the result of annotating [stmt] in [env'] with [return_type].
+
+    {4 Example}
 
-  {3 TypingRule.Catcher}
+
+  {3 TypingRule.CatcherSome}
 
     {4 Rule}
-    <description>
+    All of the following applies:
+     - [ty] has the structure of an exception type;
+    - [name_opt] gives a name [name];
+    - [name] is not already declared in [env];
+    - [name] is annotated with [ty] in [env];
+    - [env'] is [env];
+    - [new_stmt] is the result of annotating [stmt] in [env'] with [return_type].
 
     {4 Example}
-    <minimal example>
+
 
 {2
 Typing of Function Calls
 }
-[annotate_call pos name env args named_args] annotates the call to function [name]
-with arguments [args] and parameters [named_args].
+[annotate_call loc en name args eqs call_type] annotates the call to function
+[name] with arguments [args] and parameters [eqs]. Formally, [annotate_call loc
+en name args eqs call_type] is [(name1, args, eqs2, ret_ty1)] or an error is
+raised and one of the following applies:
 
-  {3 TypingRule.FCall}
+  {3 TypingRule.FCallBadArity}
 
     {4 Rule}
-    <description>
+    All of the following applies:
+    - [caller_arg_types, arg1] is the result of annotating [args] in [env];
+    - [extra_nargs, name1, callee_arg_types, ret_ty] is the result of trying to
+      find the name [name] with [caller_arg_types] in [env];
```

Comment:
I find the following a bit more understandable, but I'm not sure it's that better:

```suggestion
    - [name] is bound in [env] to a function with argument types [callee_arg_types];
```